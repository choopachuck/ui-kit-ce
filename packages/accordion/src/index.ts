export * from './Accordion'
export * from './AccordionItem'
export type {
  AccordionIconProps,
  AccordionComponentsConfig,
} from './components'
