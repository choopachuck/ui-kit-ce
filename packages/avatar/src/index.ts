export * from './Avatar'
export * from './AvatarGroup'
export * from './types'
export type {
  AvatarGroupExtraProps,
  AvatarGroupComponentsConfig,
} from './components'
export * from './classes'
export * from './constants'
