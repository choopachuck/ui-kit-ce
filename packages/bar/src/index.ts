export * from './Bar'
export * from './BarButton'
export * from './BarDate'
export * from './BarDivider'
export * from './BarDropdown'
export * from './BarDropdownItem'
export * from './BarItem'
export * from './BarMenuItem'
export * from './BarSearch'
export * from './BarSelect'
export * from './constants'
export * from './interfaces'
export * from './SubBar'
