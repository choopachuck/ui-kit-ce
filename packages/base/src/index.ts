export * from '@v-uik/accordion'
export * from '@v-uik/autocomplete'
export * from '@v-uik/badge'
export * from '@v-uik/bar'
export * from '@v-uik/box'
export * from '@v-uik/breadcrumbs'
export * from '@v-uik/button'
export * from '@v-uik/button-group'
export * from '@v-uik/checkbox'
export * from '@v-uik/checkbox-group'

export * from '@v-uik/combo-box'
export type { Classes as ComboBoxClasses } from '@v-uik/combo-box'
export type { CommonProps as ComboBoxCommonProps } from '@v-uik/combo-box'

export * from '@v-uik/common'
export * from '@v-uik/container'
export * from '@v-uik/date-picker'
export * from '@v-uik/divider'
export * from '@v-uik/drawer'
export * from '@v-uik/dropdown'
export * from '@v-uik/dropdown-menu'
export * from '@v-uik/file-uploader'
export * from '@v-uik/grid'
export * from '@v-uik/hooks'
export * from '@v-uik/input'

export * from '@v-uik/input-helper-text'
export type { Classes as InputHelperTextClasses } from '@v-uik/input-helper-text'

export * from '@v-uik/input-label'
export type { Classes as InputLabelClasses } from '@v-uik/input-label'

export * from '@v-uik/input-number'
export * from '@v-uik/input-password'
export * from '@v-uik/label-control'
export * from '@v-uik/link'
export * from '@v-uik/list'
export * from '@v-uik/masked-input'
export * from '@v-uik/modal'
export * from '@v-uik/notification'
export * from '@v-uik/popup'
export * from '@v-uik/portal'
export * from '@v-uik/progress'
export * from '@v-uik/radio'
export * from '@v-uik/radio-group'

export * from '@v-uik/select'
export type { Classes as SelectClasses } from '@v-uik/select'

export * from '@v-uik/slider'
export type { Classes as SliderClasses } from '@v-uik/slider'

export * from '@v-uik/stepper'
export * from '@v-uik/switch'
export * from '@v-uik/table'
export * from '@v-uik/tabs'

export * from '@v-uik/tag'
export type { CommonProps as TagCommonProps } from '@v-uik/tag'

export * from '@v-uik/textarea'
export * from '@v-uik/theme'

export * from '@v-uik/tooltip'
export type { Classes as TooltipClasses } from '@v-uik/tooltip'

export * from '@v-uik/typography'
export * from '@v-uik/underlay'
export * from '@v-uik/utils'
export * from '@v-uik/labelled'
export * from '@v-uik/avatar'

/**
 * Заглушка, не используется
 * @deprecated
 */
export type Classes = unknown
/**
 * Заглушка, не используется
 * @deprecated
 */
export type CommonProps = unknown
