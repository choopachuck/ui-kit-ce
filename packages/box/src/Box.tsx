import * as React from 'react'

/**
 * Для типизации внешних компонентов.
 */
export type PolymorphicComponentProps<E extends React.ElementType, P> = P &
  BoxProps<E>

/**
 * Для типизации внешних компонентов.
 */
export type PolymorphicComponent<P, D extends React.ElementType = 'div'> = <
  E extends React.ElementType = D
>(
  props: PolymorphicComponentProps<E, P>
) => React.ReactElement | null

export type BoxOwnProps<E extends React.ElementType = React.ElementType> = {
  /**
   * HTML-тег, отображаемый компонентом.
   */
  as?: E
}

export type BoxProps<E extends React.ElementType> = BoxOwnProps<E> &
  Omit<
    React.ComponentProps<E>,
    keyof BoxOwnProps | 'onPointerEnterCapture' | 'onPointerLeaveCapture'
  >

const defaultElement = 'div'

/**
 * Polymorphic React components.
 *
 * @see https://github.com/kripod/react-polymorphic-box
 * @see https://blog.andrewbran.ch/polymorphic-react-components/
 * @see https://www.benmvp.com/blog/forwarding-refs-polymorphic-react-component-typescript/
 */
export const Box: <E extends React.ElementType = typeof defaultElement>(
  props: BoxProps<E>
) => React.ReactElement | null = React.forwardRef(function Box(
  props: BoxOwnProps,
  ref: React.Ref<Element>
) {
  const Element = props.as || defaultElement

  return <Element ref={ref} {...props} as={undefined} />
})
