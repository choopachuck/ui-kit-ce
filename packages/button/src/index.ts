export * from './Button'
export * from './useButtonReset'
