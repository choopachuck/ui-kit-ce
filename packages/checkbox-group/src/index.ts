export * from './CheckboxGroup'
export * from './CheckboxGroupContext'
export * from './interfaces'
