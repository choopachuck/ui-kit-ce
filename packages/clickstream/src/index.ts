export * from './ClickStreamProvider'
export type {
  ClickStreamBaseData,
  ClickStreamContextProps,
  ClickStreamElementProperty,
  ClickStreamElementType,
  ClickStreamEventBatch,
  ClickStreamEventBatchItem,
  ClickStreamEventData,
  ClickStreamEventMeta,
  ClickStreamProviderProps,
} from './types'
