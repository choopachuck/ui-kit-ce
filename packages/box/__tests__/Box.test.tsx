import * as React from 'react'
import { Box } from '../src'
import { render } from '@testing-library/react'

it('render component with default tag div and data-test attribute', () => {
  const { container } = render(<Box data-test="cool!" />)

  expect(container).toMatchSnapshot()
})

it('render component with tag h1', () => {
  const { container } = render(<Box as="h1" />)

  expect(container).toMatchSnapshot()
})

it('use ref for change input value', async () => {
  const CustomInput = () => {
    const ref = React.useRef<HTMLInputElement | null>(null)

    React.useEffect(() => {
      if (ref.current) {
        ref.current.value = 'test'
      }
    })

    return <Box ref={ref} as="input" />
  }
  const { findByRole } = render(<CustomInput />)

  const input = (await findByRole('textbox')) as HTMLInputElement

  expect(input.value).toBe('test')
})
