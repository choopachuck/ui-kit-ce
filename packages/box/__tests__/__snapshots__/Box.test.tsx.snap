// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component with default tag div and data-test attribute 1`] = `
<div>
  <div
    data-test="cool!"
  />
</div>
`;

exports[`render component with tag h1 1`] = `
<div>
  <h1 />
</div>
`;
